src/main/java/nl/novi/serviceconnect/controllers/QueryController.java

```diff
@@ -24,4 +26,11 @@ public ResponseEntity<List<CategoryServiceCountDto>> countServicesPerCategory()
         List<CategoryServiceCountDto> result = queryService.countServicesPerCategory();
         return ResponseEntity.ok(result);
     }
+
+    @GetMapping("/search")
+    public ResponseEntity<List<ServiceOutputDto>> searchServices(@RequestParam(required = false) Long categoryId,
+                                                                 @RequestParam(required = false) Double minPrice,
+                                                                 @RequestParam(required = false) Double maxPrice) {
```

Comment:
Ik zou ze niet "Required" maken, maar optioneel. Je moet dan nog wel checken of ze aanwezig zijn en vervolgens dynamisch een query uitvoeren